pkg/controller/vitesskeyspace/reconcile_topo.go

```diff
@@ -55,6 +55,17 @@ func (r *ReconcileVitessKeyspace) reconcileTopology(ctx context.Context, vtk *pl
 	}
 	defer ts.Close()
 
+	if *vtk.Spec.TopologyReconciliation.PruneShards {
+		result, err := r.pruneShards(ctx, vtk, ts)
+		resultBuilder.Merge(result, err)
+	}
+
+	return resultBuilder.Result()
+}
+
+func (r *ReconcileVitessKeyspace) pruneShards(ctx context.Context, vtk *planetscalev2.VitessKeyspace, ts *toposerver.Conn) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+
 	// Get list of keyspaces in topo.
```

Comment:
Not your fault, but can you fix this comment? It should say shards, not keyspaces.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/26/commits/bd53c9410b840112368abb39f270307cc0b39209.